cli-quarkus/src/main/java/org/eclipse/hono/cli/app/NorthBoundApis.java

```diff
@@ -0,0 +1,243 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+
+package org.eclipse.hono.cli.app;
+
+import java.util.HashMap;
+import java.util.Optional;
+
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.kafka.common.security.scram.ScramLoginModule;
+import org.apache.kafka.common.security.scram.internals.ScramMechanism;
+import org.eclipse.hono.application.client.ApplicationClient;
+import org.eclipse.hono.application.client.MessageContext;
+import org.eclipse.hono.application.client.amqp.ProtonBasedApplicationClient;
+import org.eclipse.hono.application.client.kafka.impl.KafkaApplicationClientImpl;
+import org.eclipse.hono.cli.util.ConnectionOptions;
+import org.eclipse.hono.cli.util.PropertiesVersionProvider;
+import org.eclipse.hono.client.amqp.config.ClientConfigProperties;
+import org.eclipse.hono.client.amqp.connection.HonoConnection;
+import org.eclipse.hono.client.kafka.CommonKafkaClientConfigProperties;
+import org.eclipse.hono.client.kafka.consumer.MessagingKafkaConsumerConfigProperties;
+import org.eclipse.hono.client.kafka.producer.CachingKafkaProducerFactory;
+import org.eclipse.hono.client.kafka.producer.MessagingKafkaProducerConfigProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.quarkus.runtime.ShutdownEvent;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import picocli.CommandLine;
+import picocli.CommandLine.Model.CommandSpec;
+import picocli.CommandLine.ParameterException;
+
+/**
+ * Commands for interacting with Hono's north bound APIs.
+ *
+ */
+@Singleton
+@CommandLine.Command(
+        name = "app",
+        description = { "A client for interacting with Hono's north bound API endpoints." },
+        mixinStandardHelpOptions = true,
+        versionProvider = PropertiesVersionProvider.class,
+        sortOptions = false,
+        subcommands = { TelemetryAndEvent.class, CommandAndControl.class })
+public class NorthBoundApis {
+
+    private static final Logger LOG = LoggerFactory.getLogger(NorthBoundApis.class);
+
+    private static final String SANDBOX_AMQP_USER = "consumer@HONO";
+    private static final String SANDBOX_AMQP_PWD = "verysecret";
+    private static final String SANDBOX_KAFKA_USER = "hono";
+    private static final String SANDBOX_KAFKA_PWD = "hono-secret";
+
+    @CommandLine.Mixin
+    ConnectionOptions connectionOptions;
+
+    @CommandLine.Option(
+            names = {"--amqp"},
+            description = {
+                "Connect to the AMQP 1.0 based API endpoints",
+                "If not set, the Kafka based endpoints are used by default" },
```

Comment:
Seeing Kafka as default here was a surprise to me. In my first test I omitted this parameter while trying to connect to an AMQP messaging network. Only after several seconds there were some Kafka errors being printed out.
I think I would prefer a more explicit definition of the messaging type here, e.g. with a mandatory `--type kafka` parameter (or having to use `--kafka` ).
At least there could be a message like `Connecting to Kafka at host:port.`.

Reply:
So you expected to connect to AMQP based infrastructure *by default*, right? I guess, then, that you only expect that because you have been working with Hono for so long already. New users who read in the documentation and on the website that Kafka based messaging infrastructure is the default would be wondering why they need to specify an option for using the default, wouldn't they?

Reply:
We can, however, print something to standard error, indicating the type of messaging infrastructure ...

Reply:
> New users who read in the documentation and on the website that Kafka based messaging infrastructure is the default would be wondering why they need to specify an option for using the default, wouldn't they?

Yes, so this is associated with #3218 then.

Reply:
The changes I have pushed recently also include printing the type of messaging infra to standard error ...